### DISCUSSION TITLE
Proposal: `#align-indent!` instead of `@zero_indent` and `@aligned_indent`

### DISCUSSION THREAD

**Discussion by vhyrro** [nvim-treesitter/nvim-treesitter#2509] [Ideas]
Hi!

I'm really excited to hear that TS-based indentation is getting some updates! I did realize something though - `@zero_indent` and `@aligned_indent` seem a little redundant, am I wrong?

Wouldn't it be more sensible to perhaps allow `(#align-indent!)` instead? It would allow either a static value (e.g. `(#align-indent! 0)` would be the equivalent of `@zero_indent`) or a capture (e.g. `(#align-indent! @my-capture)` would indent to the start of `my-capture`).

Is there any reason that the current implementation is used vs directives? Thanks!

**Comment by kyazdani42** [comment]
Not really, but would probably be slower to run a custom capture than just tagging nodes.

**Reply by theHamsta** [reply]
And it would be interesting on what you would run the directive on. Some captures already support setting arguments via set. I added a note that we probably want fixed-indent + number instead of zero_indent but I don't see the connection with aligned_indent

**Reply by vhyrro** [reply]
@theHamsta perhaps it's just my flawed interpretation of what `aligned_indent` does (not all that much info in the docs lol), but my thought was to make aligning to any node through a capture possible, versus doing whatever "hardcoded" aligning `aligned_indent` does.

Would be handy for scenarios like this (using markdown as a base cause it has syntax highlight):
```markdown
# My Heading
  Text <- text gets aligned directly under the "My Heading" node.
```

As for what you'd run the directive on - you're right, i didn't really think of that lol

I would assume:
```
(
	(atx1_marker
		title: (text) @_title
	)
	(#align-indent! @_title)
)
```
But yeah it does feel a little wrong not only because of the anonymous `()` but also because where would you place e.g. an `@indent`/`@dedent`/whatever tag? Would they even apply here? Oh well, was worth toying with the idea tho.

Thanks for the replies :)

**Reply by theHamsta** [reply]
@vhyrro you can already specify an relative offset for aligned_indent it's just a bit hidden and you have to specify it via `set!`

**Reply by vhyrro** [reply]
Ooh fancy! Thanks a bunch for helping me out here. I'll look into that then :)

With the `#set!` invocation things make a lot more sense now, you specify the `@aligned_indent` capture, then say how you want to align, nice.
